tests/test_yamls/intermediate_bucket.yaml

```diff
@@ -0,0 +1,21 @@
+name: intermediate-bucket
```

Comment:
Can we just use `managed_job_with_storage.yaml` instead? Refer: https://github.com/skypilot-org/skypilot/blob/914328acb8269d79e304ad891f84d220e077565c/tests/test_smoke.py#L3283

Reply:
@zpoint pinging on this - it might be good to reuse `managed_job_with_storage.yaml` instead of having to add this file. Would that be possible?

Reply:
I am using `examples/managed_job_with_storage.yaml` for the job setup and `tests/test_yamls/use_intermediate_bucket_config.yaml` for the environment setup. This reuses the job logic, so just verify the intermediate_bucket logic.